### PR TITLE
feat(tech lifecycle) add Maven content in tech-lifecycle chapter 

### PR DESCRIPTION
**Pull request by dduportal**
Depends on #47 .

This PRs adds the content of the section "tech lifecycle".

Initially imported from last year (2020) content, this section is reworked to provide a smoother content.

### CODE REVIEW

File: content/chapitres/project-lifecycle.adoc [hunk 1]
```diff
@@ -1,4 +1,779 @@
...
+
+C'est un fichier `*.jar` identifié par... groupId, artifactId et version.
+
+== Exemple Maven : plugin Spring Boot 1/2
+
+=> C'est à vous !
+
+* *But :* Configurer le plugin Spring Boot pour Maven
+** Vous devez mettre à jour le fichier `pom.xml` en suivant la
+https://docs.spring.io/spring-boot/docs/2.5.6/maven-plugin/reference/htmlsingle/[documentation du plugin Maven Spring Boot]
```
Comment (jlevesy) [inline content/chapitres/project-lifecycle.adoc]:
```suggestion
https://docs.spring.io/spring-boot/docs/2.6.0/maven-plugin/reference/htmlsingle/[documentation du plugin Maven Spring Boot]
```

For the sake of being picky 😅

File: content/chapitres/project-lifecycle.adoc [hunk 2]
```diff
@@ -1,4 +1,779 @@
...
+== Exercice : Prévisualiser l'application
+
+* Dans un second terminal de Gitpod, affichez la page web de l'application avec les commandes suivantes :
+** `gp url 8080` pour afficher l'URL publique de l'application correspondant au port `8080` local
+** `gp preview "$(gp url 8080)/"` pour prévisualiser le "endpoint" `/` dans un navigateur local
+
+* La page d'accueil doit afficher HTTP/404
+
+* Trouvez la page des menus qui doit répondre `[]` (liste vide en JSON)
+** 💡 `src/\*/controller/*.java`
```
Comment (jlevesy) [inline content/chapitres/project-lifecycle.adoc:729]:
```suggestion
Dans un second terminal de Gitpod, listez les menus enregistrés dans le menu-server comme présenté dans le chapitre précédent en utilisant curl.
```

Nit: We're not really talking about page, but API endpoint?
Also, they already know about this `GET /menus` endpoint from the previous chapter, as we already make them play with curl. 
Soooo, I would say we don't really need to mention gitpod public URLs, only using curl simplifies the whole thing?

WDYT?

**Review by jlevesy — COMMENTED** [review]
Sounds good, minor nits. Happy to talk about it :)


Thanks for the work

File: content/chapitres/project-lifecycle.adoc (hunk 1 above)
Comment (dduportal) [reply content/chapitres/project-lifecycle.adoc]:
```suggestion
https://docs.spring.io/spring-boot/docs/{springboot_version}/maven-plugin/reference/htmlsingle/[documentation du plugin Maven Spring Boot]
```

Good catch 👍

File: content/chapitres/project-lifecycle.adoc (hunk 2 above)
Comment (dduportal) [reply content/chapitres/project-lifecycle.adoc:729]: I'm not at ease with letting them only use `curl`. Being able to visualize through a webbrowser makes it more real.
We can still recommend `curl -v http://localhost:8080` next to it though WDYT?

Comment (jlevesy) [reply content/chapitres/project-lifecycle.adoc:729]: If we do it here, we need to do it also in the previous part then, would be less confusing for the students? (like why we need to expose now, and not before).

Comment (dduportal) [reply content/chapitres/project-lifecycle.adoc:729]: yep, make sense. Let's add a new PR for this after. Sounds good for you?

**Pull request merged**